src/libraries/Native/Unix/System.Native/ios/pal_sys.m

```diff
@@ -0,0 +1,44 @@
+
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include "pal_sys.h"
+#import <Foundation/Foundation.h>
+
+void SystemNative_Log (uint8_t* buffer, int32_t length)
+{
+    // COOP: no managed memory access: any mode.
+    NSString *msg = [[NSString alloc] initWithBytes: buffer length: length encoding: NSUTF16LittleEndianStringEncoding];
+    if (length > 4096)
+    {
+        // Write in chunks of max 4096 characters; older versions of iOS seems to have a bug where NSLog may hang with long strings (!).
+        // https://github.com/xamarin/maccore/issues/1014
+        const char* utf8 = [msg UTF8String];
```

Comment:
According to the issue it reproduces even on iOS 11 which is higher than the minimal version we support as far as I understand.

Reply:
This is Objective C?  Do we have any other use of Objective C in System.Native?  Is it required to build this for iOS?

Reply:
It's probably possible to avoid it  but does it matter? It's iOS only and we most likely will need to bring more iOS specific pieces here for other APIs, most likely for globalization stuff

Reply:
> It's iOS only

How should I interpret that in conjunction with Marek's comments about the interop .cs file being moved to a general mac folder?

> does it matter?

I don't know.  I don't have a lot of experience with Objective C and whatever toolset requirements it might or might not place on the repo.

Reply:
> How should I interpret that in conjunction with Marek's comments about the interop .cs file being moved to a general mac folder?

Replied on the other thread, it should be some generic apple folder, not mac.

> Objective C and whatever toolset requirements it might or might not place on the repo.

No dependencies, it's part of the default XCode toolchain you need on Mac for everything.

Reply:
Also, I guess next will be `Environment.GetFolderPath` https://github.com/xamarin/xamarin-macios/blob/3f0985ecac82b00aada1a41a464b15d1f58b6744/runtime/xamarin-support.m#L173-L182 so objc will be unavoidable (to call the native API).